src/ui/components/DeckBuilder/useDeckBuilder.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import * as AllCards from '../../../game/cards'
+import { DECK_SIZE } from '../../../game/config'
+import { pricing } from '../../../game/services/Pricing'
+import { CardType, ICard, ICrop } from '../../../game/types'
+import { isCard } from '../../../game/types/guards'
+
+interface UseDeckBuilderProps {
+  onDone: (deck: Map<ICard, number>) => void
+}
+
+export const useDeckBuilder = ({ onDone }: UseDeckBuilderProps) => {
+  const sortedCards = useMemo(() => {
+    const cards = (Object.values(AllCards) as unknown[]).filter(isCard)
+
+    const crops = cards.filter(c => c.type === CardType.CROP) as ICrop[]
+    const water = cards.filter(c => c.type === CardType.WATER)
+    const tools = cards.filter(c => c.type === CardType.TOOL)
+    const events = cards.filter(c => c.type === CardType.EVENT)
+
+    const sortedCrops = [...crops].sort(
+      (a, b) => pricing.getCropBaseValue(a) - pricing.getCropBaseValue(b)
+    )
+
+    return [...sortedCrops, ...water, ...tools, ...events]
+  }, [])
```

Comment:
Does `sortedCards` need to be computed as part of the first render, or can it can it be moved to the component level to avoid reprocessing?